src/browser/index.js

```diff
@@ -19,43 +19,67 @@
   const Gesture = require('../service/gesture');
   const SynthesisUI = require('./synthesis-ui');
   const Force = require('../service/force');
+  const version = require('../../package.json').version;
+  const {Renderer} = require('driver')
+
 
   const isReload = window.application != null;
-  // Set up a address (message bus if you like) that will be used
-  // as an address for all application components / services. This
-  // address is going to receive action and then pass it on to each
-  // application component for it handle it.
-  const address = new Address({
-    receive(action) {
-      // This is unfortunate hack to workaround gecko issues: #566, #565, #564
-      // Basic idea is that actions that need to happen in the same tick are
-      // boxed with `Force({action})`. Such actions are unboxed and render
-      // is forced at the end of the receive. This is likey to introduce some
-      // race conditions but for now that's the best we can do.
-      const isForced = action instanceof Force.Action;
-      if (isForced) {
-        action = action.action;
-      }
-
-      application.receive(action);
-      thumbnail(action);
-      pallet(action);
-      runtime(action);
-      suggestion(action);
-      keyboard(action);
-      settings(action);
-      scraper(action);
-      synthesis(action);
-      gesture(action);
-
-      // We cancel scheduled render on next animation frame as we are
-      // forceing render to happen right away.
-      if (isForced) {
-        cancelAnimationFrame(application.version);
-        application.render();
-      }
-    }
+
+
+  const application = start({
+    initial: isReload ?
+                Browser.Model(window.application.model.value) :
+                Session.restore(version, Browser.Model),
+    update: Browser.update,
+    view: (model, _) => Browser.view(model, address)
```

Comment:
We use this hack of wrapping address so that we could delegate every action to a services that were implemented in the past. In a future most services will be replaced by effects while some of the use cases will be replaced by forwarding addresses.